datafusion/src/physical_plan/expressions/in_list.rs

```diff
@@ -99,6 +124,104 @@ macro_rules! make_contains {
     }};
 }
 
+macro_rules! make_contains_primitive {
+    ($ARRAY:expr, $LIST_VALUES:expr, $NEGATED:expr, $SCALAR_VALUE:ident, $ARRAY_TYPE:ident) => {{
+        let array = $ARRAY.as_any().downcast_ref::<$ARRAY_TYPE>().unwrap();
+
+        let mut contains_null = false;
+        let values = $LIST_VALUES
+            .iter()
+            .flat_map(|expr| match expr {
+                ColumnarValue::Scalar(s) => match s {
```

Comment:
Given this code just seems to pass along the values,  wonder if it would be clearer / faster here to look for nulls using the `ScalarValue::is_null()`, perhaps like

```rust
let contains_nulls = $LIST_VALUES.iter().any(|expr| expr.is_null());
```

And that way you could stop when you hit the first one as well as potentially avoid a collect into a new `Vec`